coding/python_style_guide.rst

```diff
@@ -12,648 +13,671 @@ To request changes to these standards, please file an :ref:`RFC <decision-making
 
-``self`` MUST be used for the first argument to instance methods
-----------------------------------------------------------------
+:command:`autopep8` changes must always be validated before committing.
 
-Always use ``self`` for the first argument to instance methods.
+Style changes must be encapsulated in a distinct commit (see :ref:`git-commit-organization-logical-units` in :doc:`Workflow document <../processes/workflow>`).
 
-.. _style-guide-py-2-14:
+.. note::
 
-``cls`` MUST be used for the first argument to class methods
-------------------------------------------------------------
+   :command:`autopep8` only fixes PEP 8 issues and does not address other guildelines listed here.
 
-Always use ``cls`` for the first argument to class methods.
+.. _autopep8: https://pypi.python.org/pypi/autopep8
 
-.. _style-guide-py-2-15:
+.. _style-guide-py-layout:
 
-Names l (lowercase: el), O (uppercase: oh), I (uppercase: eye) MUST be avoided
-------------------------------------------------------------------------------
+2. Layout
+=========
 
-Never use the characters
+.. _style-guide-py-line-length:
 
-- ``l`` (lowercase letter el),
-- ``O`` (uppercase letter oh), or
-- ``I`` (uppercase letter eye) as single character variable names.
+Line Length MUST be less than or equal to 110 columns
+-----------------------------------------------------
 
-In some fonts, these characters are indistinguishable from the numerals one and zero.
-When tempted to use ``l``, use ``L`` instead.
+Limit all lines to a maximum of 110 characters.
+This conforms to the :doc:`cpp_style_guide` (see :ref:`4-6 <style-guide-cpp-4-6>`).
 
-.. _style-guide-py-inheritance:
+This differs from the `PEP 8 recommendation of 79 characters <https://www.python.org/dev/peps/pep-0008/#id19>`_.
 
-3. Designing for Inheritance
-============================
+.. _style-guide-py-implied-continuation:
 
-Always decide whether a class's methods and instance variables (collectively: "attributes") should be public or non-public.
-If in doubt, choose non-public; it's easier to make it public later than to make a public attribute non-public.
+Python's implied continuation inside parens, brackets and braces SHOULD be used for wrapped lines
+-------------------------------------------------------------------------------------------------
 
-Public attributes are those that you expect unrelated clients of your class to use, with your commitment to avoid backward incompatible changes.
-Non-public attributes are those that are not intended to be used by third parties; you make no guarantees that non-public attributes won't change or even be removed.
+The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces.
+If necessary, you can add an extra pair of parentheses around an expression, but sometimes using a backslash looks better.
+Make sure to indent the continued line appropriately. Some examples:
```

Comment:
I have a slight quibble over "Some examples" vs "An example" but I suppose it depends on whether you think there is a single code example here or many examples in one snippet.